pyfr/solvers/euler/kernels/rsolvers/exact.mako

```diff
@@ -0,0 +1,211 @@
+<%namespace module='pyfr.backends.base.makoutil' name='pyfr'/>
+<%include file='pyfr.solvers.euler.kernels.flux'/>
+
+<% gamma = c['gamma'] %>
+<% hgm = 0.5*(gamma - 1) %>
+<% grgm = gamma/(gamma - 1) %>
+<% trgm = 2/(gamma - 1) %>
+<% trgp = 2/(gamma + 1) %>
+<% gmrtg = (gamma - 1)/(2*gamma) %>
+<% gprtg = (gamma + 1)/(2*gamma) %>
+<% tgrgm = (2*gamma)/(gamma - 1) %>
+<% gmrgp = (gamma - 1)/(gamma + 1) %>
+
+<%pyfr:macro name='star_flux' params='p, ps, rs, cs, f, fd'>
+    // Star Flux, assuming covolume = 0. See Toro 2009 Eq.(4.86 - 4.87)
+    if (p <= ps)
+    {
+       fpdtype_t pr = p / ps;
+       fpdtype_t pr_g = pow(pr, ${-gprtg});
+       fd = pr_g / (rs*cs);
+       f = ${trgm}*cs*(pr_g*pr - 1);
+    }
+    else
+    {
+       fpdtype_t pbs_inv = 1 / (p + ${gmrgp}*ps);
+       fpdtype_t sapb = sqrt(${trgp} * pbs_inv / rs);
+       f = (p - ps)*sapb;
+       fd = sapb - 0.5*f*pbs_inv;
+    }
+</%pyfr:macro>
+
+<%pyfr:macro name='primitives' params='s, p, v, c'>
+    fpdtype_t invrho = 1/s[0], E = s[${nvars - 1}];
+
+    // Compute the velocities
+% for i in range(ndims):
+    v[${i}] = invrho*s[${i + 1}];
+% endfor
+
+    // Compute the pressure
+    p = ${gamma - 1}*(E - 0.5*invrho*${pyfr.dot('s[{i}]', i=(1, ndims + 1))});
+
+    // Compute the local sound speed
+    c = sqrt(${gamma}*p*invrho);
+</%pyfr:macro>
+
+<% kmax = 3 %>
+
+<%pyfr:macro name='rsolve_1d' params='ul, ur, nf'>
+    fpdtype_t vl[${ndims}], pl, cl, fsl, fdl;
+    fpdtype_t vr[${ndims}], pr, cr, fsr, fdr;
+    fpdtype_t p0, p1;
+    fpdtype_t w0[${nvars}];
+
+    // Compute the left/right primitives
+    ${pyfr.expand('primitives', 'ul', 'pl', 'vl', 'cl')};
+    ${pyfr.expand('primitives', 'ur', 'pr', 'vr', 'cr')};
+
+    // Inital pressure guess
+    fpdtype_t rl = ul[0];
+    fpdtype_t rr = ur[0];
+    fpdtype_t bpv = fmax(0, 0.125*(vl[0] - vr[0])*(rl + rr)*(cl + cr) +
+                         0.5*(pl + pr));
+    fpdtype_t pmin = fmin(pl, pr);
```

Comment:
Check convention for if we use min/max or fmin/fmax.

Reply:
I think fmin/fmax are more robust than min/max, I've had some issues with rounding on the openmp backend.